modules/core/04-channel/keeper/upgrade.go

```diff
@@ -569,22 +569,25 @@ func (k Keeper) ChanUpgradeCancel(ctx sdk.Context, portID, channelID string, err
 		return errorsmod.Wrapf(types.ErrUpgradeNotFound, "port ID (%s) channel ID (%s)", portID, channelID)
 	}
 
-	if errorReceipt.Sequence < channel.UpgradeSequence {
-		return errorsmod.Wrapf(types.ErrInvalidUpgradeSequence, "error receipt sequence (%d) must be greater than or equal to current upgrade sequence (%d)", errorReceipt.Sequence, channel.UpgradeSequence)
-	}
-
-	// an error receipt proof must be provided if the sender is not authorized to cancel upgrades.
-	if !isAuthority && len(errorReceiptProof) == 0 {
-		return errorsmod.Wrap(commitmenttypes.ErrInvalidProof, "cannot submit an empty error receipt proof unless the sender is authorized to cancel upgrades")
-	}
-
 	// if the msgSender is authorized to make and cancel upgrades AND the current channel has not already reached FLUSHCOMPLETE
 	// then we can restore immediately without any additional checks
 	// otherwise, we can only cancel if the counterparty wrote an error receipt during the upgrade handshake
 	if isAuthority && channel.State != types.FLUSHCOMPLETE {
 		return nil
 	}
 
+	// an error receipt proof must be provided if the sender is not authorized to cancel upgrades.
+	// the error receipt should also have a sequence greater than or equal to the current upgrade sequence.
+	if !isAuthority {
```

Comment:
This is a bug? As per comment above:
```
// if the msgSender is authorized to make and cancel upgrades AND the current channel has not already reached FLUSHCOMPLETE
// then we can restore immediately without any additional checks
// otherwise, we can only cancel if the counterparty wrote an error receipt during the upgrade handshake
```

If the channel state is in FLUSHCOMPLETE, regardless of whether the authority called this function, we must prove that the counterparty wrote an error receipt for a valid upgrade sequence. Otherwise the authority could cancel an upgrade while the counterparty simultaneously is provided historical proof to upgrade to open. In this case, the authority would be harming itself, but no harm in ensuring this isn't possible